packages/material-ui/src/ButtonBase/ButtonBase.js

```diff
@@ -99,82 +121,119 @@ class ButtonBase extends React.Component {
         ].join(' '),
       );
     }
-    prepareFocusVisible(button.ownerDocument);
+    return button.ownerDocument;
+  }, []);
+  const { isFocusVisible, onBlurVisible } = useIsFocusVisible(getOwnerDocument);
 
-    if (this.props.action) {
-      this.props.action({
+  React.useEffect(() => {
+    if (action) {
+      action({
```

Comment:
We've been using `useImperativeHandle` for `action` props. Was this a decision based on `focusVisible` containing a setter or can this use `useImperativeHandle` too?

Reply:
Right we can use that too. I just thought "well `action` never supported refs so no useImperativeHandle" but `useImperativeHandles` makes much more sense.

Reply:
Outdated